rsocket-core/src/main/java/io/rsocket/RSocketAdapter.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.rsocket;
+
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+/**
+ * Package private class with default implementations for use in {@link RSocket}. The main purpose
+ * is to hide static {@link UnsupportedOperationException} declarations.
+ */
+class RSocketAdapter {
+
+  private static final Mono<Void> UNSUPPORTED_FIRE_AND_FORGET =
+      Mono.error(new UnsupportedOperationException("Fire-and-Forget not implemented."));
```

Comment:
@rstoyanchev  
JFY  
There were several issues related to static exceptions.  
For example https://github.com/reactor/reactor-core/issues/1710 and https://github.com/reactor/reactor-core/issues/1872

Reply:
Yes I am aware, thanks for the reminder @rsocket/rsocket-rb and pointers. I've taken a similar approach as Netty, turning off suppressed exceptions. I think that's a perfectly legitimate trade-off, in this case there isn't much use for the Reactor debugging feature that relies on suppressed exceptions.